src/components/product.js

```diff
@@ -580,6 +610,40 @@ export default class Product extends Component {
     }
   }
 
+  onCarouselItemClick(evt, target) {
+    const selectedImageId = target.dataset.imageId;
+    const imageList = this.model.images;
+    const foundImage = imageList.find((image) => {
+      return image.id === parseInt(selectedImageId, 10);
+    });
+
+    if (foundImage) {
+      this.selectedImage = foundImage;
+    }
+    this.view.render();
+  }
+
+  onCarouselChange(toRight) {
+    const imageList = this.model.images;
+    const currentImage = imageList.find((image) => {
+      return image.id === this.currentImage.id;
+    });
+    const currentImageIndex = imageList.indexOf(currentImage);
```

Comment:
interesting approach. I _think_ it's ok. (though you'll need to replaced find with filter).